### PR TITLE
Reduce peak memory in examples_torch CI job (OOM fix)

### PR DESCRIPTION
**Pull request by ydshieh**
## What does this fix?

The `examples_torch` CI job intermittently hit exit 137 (OOM killed) under the 14,250 MB pod cgroup limit. With 4 `xdist` workers running concurrently, several test functions loaded large models that pushed total system memory over the limit.

## Fix

Per-test memory reductions:

| Test | PSS before | PSS after | Fix |
|------|:----------:|:---------:|-----|
| `test_run_wav2vec2_pretraining` | ~6,081 MB | **~807 MB** | `--max_duration_in_seconds 1.0` (seq len 1,250 → 250 frames) |
| `test_run_vit_mae_pretraining` | ~5,500 MB | **~775 MB** | `--config_overrides` to 32-hidden/2-layer encoder+decoder |
| `test_run_squad` | ~3,215 MB | **~774 MB** | `tiny-random-bert` (83 K params), 30 steps |
| `test_run_clm` | ~2,916 MB | **~860 MB** | `GPT2LMHeadModel` built from tiny config, 25 epochs |
| `test_run_swag` | ~2,447 MB | **~683 MB** | `BertForMultipleChoice` built from tiny config, 30 steps |
| `test_run_mlm` | ~2,380 MB | **~883 MB** | `tiny-random-bert`, 20 epochs, `--seed=42`, threshold 42 → 100 |
| `test_run_squad_seq2seq` | ~2,436 MB | **~632 MB** | `t5-tinier-random` (~1 M params), 30 steps |
| `test_run_ner` | ~2,297 MB | **~577 MB** | `tiny-random-bert` + `--ignore_mismatched_sizes`, 10 epochs |
| `test_run_speech_recognition_seq2seq` | ~1,887 MB | **~773 MB** | `--max_duration_in_seconds 3.0` |
| `test_run_speech_recognition_ctc_adapter` | ~1,687 MB | **~771 MB** | `--max_duration_in_seconds 3.0` |
| `test_run_glue` | ~1,652 MB | **~546 MB** | `tiny-random-bert`, 30 steps |

PSS (Proportional Set Size) is per-process memory from `/proc/{pid}/smaps_rollup` — shared pages counted proportionally, so summing across workers gives the true combined footprint.

Also applies `MemoryCleanupMixin` to `ExamplesTests` (gc + cache clear between tests) and sets all `dataloader_num_workers` / `preprocessing_num_workers` to `0`.

## Verification

Peak combined PSS across 4 workers dropped from **~14,250 MB → ~4,200 MB** — 5 consecutive clean passes with no OOM.

### CODE REVIEW

**Comment by ydshieh** [comment]
**Why two tests use `tempfile.TemporaryDirectory` (model created from scratch):**

**`test_run_swag`** — `run_swag.py` doesn't expose `--ignore_mismatched_sizes`. `tiny-random-bert` is a `BertForSequenceClassification` with a `[2, 32]` head, but `BertForMultipleChoice` expects `[1, 32]` → loading fails. Fix: create `BertForMultipleChoice(config)` from scratch so the head is correct.

**`test_run_clm`** — `sshleifer/tiny-gpt2` has `n_embd=2` (zero capacity), training does nothing (perplexity ~46k). `--config_overrides` only works with `--model_type`, not `--model_name_or_path`. Fix: create `GPT2LMHeadModel(GPT2Config(n_embd=32, ...))` from scratch.

All other tests (`squad`, `ner`, `glue`, `mlm`) can use `tiny-random-bert` directly via `--model_name_or_path` because either the head size matches or `--ignore_mismatched_sizes` is exposed by the script.

File: examples/pytorch/test_pytorch_examples.py [hunk 1]
```diff
@@ -284,25 +316,40 @@ def test_run_squad_seq2seq(self):
...
+                hidden_size=32,
+                num_hidden_layers=5,
+                num_attention_heads=4,
+                intermediate_size=37,
+            )
+            BertForMultipleChoice(config).save_pretrained(model_dir)
+            AutoTokenizer.from_pretrained("hf-internal-testing/tiny-random-bert").save_pretrained(model_dir)
+
+            testargs = f"""
+                run_swag.py
```
Comment (SunMarc) [inline examples/pytorch/test_pytorch_examples.py:335]: let's just use hf-internal-testing/tiny-random-bert as we did above no ?

**Review by SunMarc — APPROVED** [review]
Thanks ! just a nit

File: examples/pytorch/test_pytorch_examples.py (hunk 1 above)
Comment (ydshieh) [reply examples/pytorch/test_pytorch_examples.py:335]: the problem is that it fails at loading due to some shape mismatch `([1, hidden_size])` v.s. `([2, hidden_size])` for this particular "task".

Comment (ydshieh) [reply examples/pytorch/test_pytorch_examples.py:335]: `tiny-random-bert` has a classification head with shape `[2, hidden_size]` (`num_labels=2`), but `BertForMultipleChoice` for SWAG needs `[1, hidden_size]` (a single logit per choice). Since `run_swag.py` does not have an `--ignore_mismatched_sizes` flag (unlike other scripts in this PR), loading the mismatched weights would fail. A fresh `BertForMultipleChoice` is therefore instantiated with the correct head dimensions.

Comment (SunMarc) [reply examples/pytorch/test_pytorch_examples.py:335]: okay !
